rbd/snapshot_octopus.go

```diff
@@ -0,0 +1,70 @@
+// +build octopus
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <stdlib.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"unsafe"
+
+	"github.com/ceph/go-ceph/internal/retry"
+)
+
+// GetID returns the snapshot ID for the given Snapshot object.
+//
+// Implements:
+//  int rbd_snap_get_id(rbd_image_t image, const char *snapname, uint64_t *snap_id)
+func (snapshot *Snapshot) GetID() (uint64, error) {
```

Comment:
I'd really like to avoid adding methods to Snapshot type and move towards properly deprecating it. I'd only be OK with adding new methods to Snapshot if there's no other way to accomplish what we need.